runtime/lua/vim/filetype/detect.lua

```diff
@@ -869,6 +869,67 @@ function M.y(bufnr)
   vim.bo[bufnr].filetype = 'yacc'
 end
 
+function M.decl(bufnr)
+  for _, line in ipairs(getlines(bufnr, 1, 3)) do
+    if line:lower():find('^<!sgml') then
+      vim.bo[bufnr].filetype = 'sgmldecl'
+      return
+    end
+  end
+end
+
+function M.dcl(bufnr)
+  for _, line in ipairs(getlines(bufnr, 1, 3)) do
+    if line:lower():find('^<!sgml') then
+      vim.bo[bufnr].filetype = 'sgmldecl'
+      return
+    end
+  end
+  vim.bo[bufnr].filetype = 'clean'
+end
+
+function M.sgml(bufnr)
+  if getlines(bufnr, 1, 5, { concat = true }):lower():find('linuxdoc') then
```

Comment:
The `opts` argument passed to `getlines` has been removed, so this should be `table.concat(getlines(bufnr, 1, 5)` (same elsewhere).